### ISSUE TITLE
Install bug fix

### ISSUE THREAD

**Issue by creiser** [facebookresearch/NSVF#19]
Replace `_ext_src_root = "fairnr/clib"` by `_ext_src_root = os.path.abspath("fairnr/clib")` in setup.py

**Comment by MultiPath** [comment]
For PyTorch 1.4.0, it is ok. For >1.5, there seems a bug and it forbids relative reference.

**Comment by kwea123** [comment]
@creiser Are you able to build for torch>1.5 with that fix? I still encounter error with torch>1.5:
```
running build_ext
building 'fairnr.clib._ext' extension
Emitting ninja build file /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/build.ninja...
Compiling objects...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
[1/4] c++ -MMD -MF /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect.o.d -pthread -B /home/kwea123/anaconda3/envs/nerf_pl/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/intersect.cpp -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect.o -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
FAILED: /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect.o
c++ -MMD -MF /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect.o.d -pthread -B /home/kwea123/anaconda3/envs/nerf_pl/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/intersect.cpp -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect.o -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
/home/kwea123/workspace/NSVF/fairnr/clib/src/intersect.cpp:6:10: fatal error: intersect.h: No such file or directory
 #include "intersect.h"
          ^~~~~~~~~~~~~
compilation terminated.
[2/4] c++ -MMD -MF /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/octree.o.d -pthread -B /home/kwea123/anaconda3/envs/nerf_pl/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/octree.cpp -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/octree.o -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
FAILED: /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/octree.o
c++ -MMD -MF /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/octree.o.d -pthread -B /home/kwea123/anaconda3/envs/nerf_pl/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/octree.cpp -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/octree.o -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
/home/kwea123/workspace/NSVF/fairnr/clib/src/octree.cpp:6:10: fatal error: octree.h: No such file or directory
 #include "octree.h"
          ^~~~~~~~~~
compilation terminated.
[3/4] c++ -MMD -MF /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/binding.o.d -pthread -B /home/kwea123/anaconda3/envs/nerf_pl/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/binding.cpp -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/binding.o -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
FAILED: /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/binding.o
c++ -MMD -MF /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/binding.o.d -pthread -B /home/kwea123/anaconda3/envs/nerf_pl/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/binding.cpp -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/binding.o -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
/home/kwea123/workspace/NSVF/fairnr/clib/src/binding.cpp:6:10: fatal error: intersect.h: No such file or directory
 #include "intersect.h"
          ^~~~~~~~~~~~~
compilation terminated.
[4/4] /usr/local/cuda/bin/nvcc -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/intersect_gpu.cu -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect_gpu.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_61,code=sm_61 -std=c++14
FAILED: /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect_gpu.o
/usr/local/cuda/bin/nvcc -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/TH -I/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/kwea123/anaconda3/envs/nerf_pl/include/python3.7m -c -c /home/kwea123/workspace/NSVF/fairnr/clib/src/intersect_gpu.cu -o /home/kwea123/workspace/NSVF/build/temp.linux-x86_64-3.7/fairnr/clib/src/intersect_gpu.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O2 -Ifairnr/clib/include -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_ext -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_61,code=sm_61 -std=c++14
/home/kwea123/workspace/NSVF/fairnr/clib/src/intersect_gpu.cu:11:10: fatal error: cuda_utils.h: No such file or directory
 #include "cuda_utils.h"
          ^~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1539, in _run_ninja_build
    env=env)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "setup.py", line 35, in <module>
    'fairnr-train = fairseq_cli.train:cli_main'
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/setuptools/command/build_ext.py", line 79, in run
    _build_ext.run(self)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/command/build_ext.py", line 340, in run
    self.build_extensions()
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 670, in build_extensions
    build_ext.build_extensions(self)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/command/build_ext.py", line 449, in build_extensions
    self._build_extensions_serial()
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/command/build_ext.py", line 474, in _build_extensions_serial
    self.build_extension(ext)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/setuptools/command/build_ext.py", line 196, in build_extension
    _build_ext.build_extension(self, ext)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/distutils/command/build_ext.py", line 534, in build_extension
    depends=ext.depends)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 500, in unix_wrap_ninja_compile
    with_cuda=with_cuda)
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1255, in _write_ninja_file_and_compile_objects
    error_prefix='Error compiling objects for extension')
  File "/home/kwea123/anaconda3/envs/nerf_pl/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1555, in _run_ninja_build
    raise RuntimeError(message) from e
RuntimeError: Error compiling objects for extension
```
torch 1.4 is fine..

**Comment by Xharlie** [comment]
Tested with python 3.7.2, pytorch1.8.1+cuda 10.2, the bug fix works.
